wom/src/main/scala/wdl4s/wom/graph/Graph.scala

```diff
@@ -23,14 +23,28 @@ object Graph {
     * Assuming it validates, construct the Graph case class.
     */
   def validateAndConstruct(nodes: Set[GraphNode]): ErrorOr[Graph] = {
-    def goodLink(port: InputPort): ErrorOr[Unit] = {
+
+    def boolToErrorOr(bool: Boolean, msg: => String): ErrorOr[Unit] = if (bool) {
```

Comment:
You could leverage `.either(Nel(msg), ()).toValidatedNel` by using `import mouse.all._` and `import cats.syntax.either`